File: src/bill_factory.cpp
#include "include/bill_factory.h"

deposit bill_factory::build_deposit(const big_int& person_id) const {
  deposit deposit_;
  deposit_.set_person_id(person_id);
  deposit_.set_cash_size(0);
  deposit_.set_income_potential_size(deposit_.get_cash_size() / 100 * 10);
  return deposit_;
}

credit bill_factory::build_credit(const big_int& person_id) const {
  credit credit_;
  credit_.set_person_id(person_id);
  credit_.set_cash_size(100000000);
  credit_.set_accessible_cash_size(100000000);
  credit_.set_debt_size(0);
  credit_.set_payment_on_this_month(0);
  return credit_;
}

debit bill_factory::build_debit(const big_int& person_id) const {
  debit debit_;
  debit_.set_person_id(person_id);
  debit_.set_cash_size(0);
  debit_.set_cashback_potential_size(debit_.get_cash_size() / 100 * 5);
  return debit_;
}

File: MIPT_bank_app1/MIPT_bank_app/app/src/main/java/com/example/mipt_bank_app/BillFactory.java
package com.example.mipt_bank_app;


public class BillFactory {

    public Deposit buildDeposit(int person_id) {
        Deposit deposit_ = new Deposit();
        deposit_.setPersonId(person_id);
        deposit_.setCashSize(0);
        deposit_.setIncomePotentialSize(deposit_.getCashSize() / 100 * 10);
        deposit_.assignId();
        return deposit_;
    }

    public Credit buildCredit(int person_id) {
        Credit credit_ = new Credit();
        credit_.setPersonId(person_id);
        credit_.setCashSize(100000000);
        credit_.setAccessibleCashSize(100000000);
        credit_.setDebtSize(0);
        credit_.setPaymentOnThisMonth(0);
        credit_.assignId();
        return credit_;
    }

    public Debit buildDebit(int person_id) {
        Debit debit_ = new Debit();
        debit_.setPersonId(person_id);
        debit_.setCashSize(0);
        debit_.setCashbackPotentialSize(debit_.getCashSize() / 100 * 5);
        debit_.assignId();
        return debit_;
    }
}
